doc/server_side_auth.md

```diff
@@ -0,0 +1,57 @@
+Server-side Authentication
```

Comment:
This is slightly misleading as we are not authenticating the server but we are exposing how the client is authenticated to the server.

How about "Server-side client authentication API".